### PR TITLE
feat: Initial /search req/resp stub

### PR DESCRIPTION
**Pull request by adilansari**
/search interface with stubbed response.

### Request
```
{
    "q": "tunnel",
    "search_fields": [
        "first_name",
        "last_name"
    ],
    "filter": {
        "last_name": "Steve"
    },
    "facet": {
        "facet stat": 0
    },
    "sort": [
        {
            "salary": "$asc"
        }
    ],
    "fields": [
        "employment",
        "history"
    ]
}
```

### Response
```
{
    "result": {
        "hits": [
            {
                "meta": {}
            }
        ],
        "facets": {
            "myField": {
                "counts": [
                    {
                        "count": 32,
                        "value": "adidas"
                    }
                ],
                "stats": {
                    "avg": 40,
                    "count": 50
                }
            }
        },
        "meta": {
            "found": 1234,
            "page": {
                "current": 2,
                "per_page": 10
            }
        }
    }
}
```

### CODE REVIEW

**Comment by adilansari** [comment]
How should we merge the `submodule changes`?

File: api/server/v1/marshaler.go [hunk 1]
```diff
@@ -131,6 +131,52 @@ func (x *ReadRequest) UnmarshalJSON(data []byte) error {
...
+			if err := jsoniter.Unmarshal(value, &x.Db); err != nil {
+				return err
+			}
+		case "collection":
+			if err := jsoniter.Unmarshal(value, &x.Collection); err != nil {
+				return err
+			}
+		case "search_fields":
+			var fields []string
+			if err := jsoniter.Unmarshal(value, &fields); err != nil {
```
Comment (himank) [inline api/server/v1/marshaler.go:152]: directly unmarshal into x.SearchFields? `jsoniter.Unmarshal(value, &x.SearchFields)`

File: server/services/v1/query_runner.go [hunk 2]
```diff
@@ -95,6 +95,15 @@ func (f *QueryRunnerFactory) GetStreamingQueryRunner(r *api.ReadRequest, streami
 	}
 }
 
+// GetSearchStreamQueryRunner for executing Search
+func (f *QueryRunnerFactory) GetSearchStreamQueryRunner(r *api.SearchRequest, streaming SearchStreaming) *SearchQueryRunner {
```
Comment (himank) [inline server/services/v1/query_runner.go:99]: `GetSearchQueryRunner `

File: server/services/v1/query_runner.go [hunk 3]
```diff
@@ -498,6 +507,48 @@ func (runner *StreamingQueryRunner) iterate(ctx context.Context, reader RowReade
...
+	*BaseQueryRunner
+
+	req       *api.SearchRequest
+	streaming SearchStreaming
+}
+
+func (runner *SearchQueryRunner) Run(ctx context.Context, tx transaction.Tx, tenant *metadata.Tenant) (*Response, context.Context, error) {
+	log.Debug().Msg("Search request:" + runner.req.String())
+
+	resp := api.SearchResponse{
```
Comment (himank) [inline server/services/v1/query_runner.go:521]: remove the dummy resp, just return `nil, ctx, api.Errorf(Code_METHOD_NOT_ALLOWED, "not implemented)` You may need to also add `Code_METHOD_NOT_ALLOWED` in errors.go.

**Comment by himank** [comment]
> How should we merge the `submodule changes`?

`git submodule update --remote --merge` to pull the latest commit and then git add

**Review by himank — APPROVED** [review]

**Pull request merged**